app/CineHub/app/src/main/java/org/cinehub/api/result/OnFailureCallback.java

```diff
@@ -0,0 +1,15 @@
+package org.cinehub.api.result;
+
+/**
+ * Callback to be called on failure.
+ * @param <T> type of the value to be passed to the callback
+ * @author Jkutkut
+ */
+public interface OnFailureCallback<T> {
```

Comment:
As this interface is meant to be used as a lambda expression, it should have a `@FunctionalInterface` annotation on top of the declaration